### PR TITLE
Fix bug 1624557: Terminology presentation popup

### PR DESCRIPTION
**Pull request by mathjazz**
This is a continuation of #1608 and a final piece of the [Terminology Presenation](https://github.com/mozilla/pontoon/blob/master/specs/0101-terminology-presentation.md) feature. It's branched off of #1608, on top of which it only adds [one new commit](https://github.com/mozilla/pontoon/commit/cabc4d01c868cb126bd98c651c73db6f34de3025), which implements term highlighting in the source string and the term popup.

**TESTING**
The patch is deployed to stage. I've put the `Usage` column of the [imported Mozilla's terms](https://mozilla-pontoon-staging.herokuapp.com/a/terminology/term/) into a [`terminology.ftl`](https://github.com/mathjazz/pontoon-test/commit/feb8a1521f4fb8fd0b47c05ef475acb5eb01cf05) file, which should help with real-life testing of term matching in the strings. See:
https://mozilla-pontoon-staging.herokuapp.com/fr/pontoon-test-2/terminology.ftl/
https://mozilla-pontoon-staging.herokuapp.com/de/pontoon-test-2/terminology.ftl/

### CODE REVIEW

File: frontend/src/core/term/withTerms.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+export function markTerms(base: string, terms: TermState) {
+    if (terms.fetching) {
+        return base;
+    }
+
+    if (!terms.terms) {
+        return base;
+    }
```
Comment (adngdb) [inline frontend/src/core/term/withTerms.js:18]: Might as well write

```js
    if (terms.fetching || !terms.terms) {
        return base;
    }
```

right? :innocent:

File: frontend/src/modules/entitydetails/components/TermsPopup.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/* @flow */
```
Comment (adngdb) [inline frontend/src/modules/entitydetails/components/TermsPopup.js:1]: Why is this file not in `modules/terms`?

File: pontoon/terminology/models.py [hunk 3]
```diff
@@ -1,31 +1,21 @@
...
-            else:
-                if re.search("".join(terms_words), "".join(source_words), flags):
-                    term_pks.append(term.pk)
-
-        return self.filter(pk__in=term_pks)
+
+            if re.search(term_text, string, flags):
+                terms.append(term)
+
+        return terms
```
Comment (adngdb) [inline pontoon/terminology/models.py:18]: What's this? I don't quite understand the previous code, but wasn't it related to tokenizing? Why did you remove it?

File: frontend/src/modules/entitydetails/components/TermsPopup.js (hunk 2 above)
Comment (mathjazz) [reply frontend/src/modules/entitydetails/components/TermsPopup.js:1]: I've put the files specific for the Terms panel into modules/terms (we use the same logic for the history, machinery, otherlocales and teamcomments panels). Since the popup is not part of the panel, I've put it in modules/entitydetails, where it belongs UI-wise.

File: frontend/src/core/term/withTerms.js (hunk 1 above)
Comment (mathjazz) [reply frontend/src/core/term/withTerms.js:14]: Good point. Updated along with several other instances across different modules.

File: pontoon/terminology/models.py (hunk 3 above)
Comment (mathjazz) [reply pontoon/terminology/models.py:18]: I should have explained it, thanks for asking. :-)

We no longer tokenize strings into words and then search for matches within tokes. Instead we just search for substrings within the string that start with the term.

That greatly simplifies the code (because we don't need a special code path for multi-word terms) and works equally good for within our test file with 200+ strings (the same terms are matched for every string).
